unstructured/partition/pdf.py

```diff
@@ -544,6 +573,12 @@ def _partition_pdf_or_image_local(
         process_file_with_pdfminer,
     )
 
+    if _is_max_pages_exceeded(filename=filename, file=file, max_pages=max_pages):
+        raise PdfMaxPagesExceededError(
+            f"Maximum number of PDF file pages exceeded - "
+            f"pages={_get_PDF_page_number(filename=filename, file=file)}, maximum={max_pages}."
```

Comment:
Hmm we'd need to avoid opening and reading the file twice - so we need to rethink the design here.

I know I've asked for `_is_max_pages_exceeded -> bool` but to have the error with nice message we'd need to introduce only one function e.g. `check_max_pages_exceeded` that won't return any value but might raise the `PdfMaxPagesExceededError`.

Sorry for the inconvenience - I wasn't conscious the number of pages won't be available here.